### PR TITLE
adding unit tests for sdk types

### PR DESCRIPTION
**Pull request by cercena**

### CODE REVIEW

**Review by FabianoEger — COMMENTED** [review]
@cercena please, resolve conflicts

File: packages/tcore-sdk/src/Types/Helpers/removeNullValues.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  test("removes null value", () => {
+    const data = { key: null };
+    const process = removeNullValues(data);
+    expect(process).toBeInstanceOf(Object);
+  });
+
+  test("removes undefined value", () => {
+    const data = { key: undefined };
+    const process = removeNullValues(data);
+    expect(process).toBeInstanceOf(Object);
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Helpers/removeNullValues.test.ts:19]: Same thing as above here

File: packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  test("check optional field", () => {
+    const data = {
+      connection_id: "connection_id",
+      device_id: "1",
+      timestamp: 0,
+      title: "title",
+    };
+
+    const parsed = zLiveInspectorMessage.parse(data);
+    expect(parsed.content).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts:56]: `toBeUndefined` is not being invoked

File: packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+});
+
+describe("zLiveInspectorMessageCreate", () => {
+  test("parses simple object", () => {
+    const data = {
+      content: "any",
+      title: "title",
+    };
+    const parsed = zLiveInspectorMessageCreate.parse(data);
+    expect(parsed.title).toEqual("title");
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts:67]: You should probably use expect to see if the other fields are there as well

File: packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  test("parses simple object", () => {
+    const data = {
+      connection_id: "connection_id",
+      content: "any",
+      device_id: "string",
+      timestamp: 0,
+      title: "title",
+    };
+    const parsed = zLiveInspectorMessage.parse(data);
+    expect(parsed.title).toEqual("title");
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/LiveInspector/LiveInspector.types.test.ts:28]: You should probably use expect to see if the other fields are there as well

File: packages/tcore-sdk/src/Types/Log/Log.types.test.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+describe("zLogCreate", () => {
+  test("parses simple object", () => {
+    const data = {
+      timestamp: new Date(),
+      message: "message",
+      error: true,
+    };
+    const parsed = zLogCreate.parse(data);
+    expect(parsed.message).toEqual("message");
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Log/Log.types.test.ts:12]: You should probably use expect to see if the other fields are there as well

File: packages/tcore-sdk/src/Types/Log/Log.types.test.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  test("parses simple object", () => {
+    const data = [
+      {
+        timestamp: new Date(),
+        message: "message",
+        error: true,
+      },
+    ];
+    const parsed = zLogList.parse(data);
+    expect(parsed[0].message).toEqual("message");
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Log/Log.types.test.ts:63]: You should probably use expect to see if the other fields are there as well

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+  zPluginStorageItemSet,
+  zPluginModule,
+  zPlugin,
+  zModuleMessageOptions,
+  zPluginModuleListItem,
+  zPluginModuleList,
+} from "./Plugin.types";
+
+describe("zPluginModuleIDCombo", () => {
+  test("invalid id", () => {
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:21]: Also add a test with it working

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+describe("zPluginType", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginType.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
+        "Invalid enum value. Expected 'action-trigger' | 'action-type' | 'database' | 'decoder' | 'encoder' | 'filesystem' | 'hook' | 'navbar-button' | 'page' | 'service' | 'sidebar-button' | 'sidebar-footer-button'"
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:40]: Don't add all the values in test, it's going to be difficult to change it later on

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+  });
+});
+
+describe("zPluginPublisher", () => {
+  test("check optional field", () => {
+    const data = {
+      name: "name",
+    };
+    const parsed = zPluginPublisher.parse(data);
+    expect(parsed.domain).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:52]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+  });
+});
+
+describe("zPluginInstallOptions", () => {
+  test("accepts empty object", () => {
+    const data = {};
+    const parsed = zPluginInstallOptions.parse(data);
+    expect(parsed.start).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:60]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+});
+
+describe("zPluginPermission", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginPermission.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:73]: Don't add all the values in test, it's going to be difficult to change it later on

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+  });
+});
+
+describe("zPluginManifestCliOption", () => {
+  test("check optional field", () => {
+    const data = {
+      flags: "flags",
+    };
+    const parsed = zPluginManifestCliOption.parse(data);
+    expect(parsed.description).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:86]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+  test("error if malformed array", () => {
+    const data = {
+      name: "name",
+      types: ["a"],
+    };
+    try {
+      zPluginPackageTCore.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.types).toStrictEqual([
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:116]: Don't add all the values in test, it's going to be difficult to change it later on

File: packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+    const data = {
+      state: "state",
+      message: 0,
+    };
+    try {
+      zPluginModule.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.state).toStrictEqual([
+        "Invalid enum value. Expected 'idle' | 'stopping' | 'starting' | 'started' | 'stopped'",
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts:157]: Don't add all the values in test, it's going to be difficult to change it later on

File: packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+  test("check default values for optional fields", () => {
+    const data = {
+      plugin_folder: "string",
+    };
+    const parsed = zSettings.parse(data);
+    //expect(parsed.plugin_auto_update_check_time).toEqual(null);
+    expect(parsed.port).toEqual(8888);
+    //expect(parsed.telemetry).toEqual(false);
+    expect(parsed.version).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts:21]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+  });
+});
+
+describe("zSettingsEdit", () => {
+  test("parses simple object", () => {
+    const data = {};
+    const parsed = zSettingsEdit.parse(data);
+    expect(parsed.plugin_folder).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts:29]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+describe("zSettingsMetadata", () => {
+  test("parses simple object", () => {
+    const data = {
+      database_plugin_disabled: true,
+      plugin_folder_disabled: true,
+      port_disabled: true,
+    };
+    const parsed = zSettingsMetadata.parse(data);
+    expect(parsed.port_disabled).toBeTruthy;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Settings/Settings.types.test.ts:41]: `toBeTruthy` not being invoked

File: packages/tcore-sdk/src/Types/Statistic/Statistic.types.test.ts [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+    const parsed = zStatisticCreate.parse(data);
+    expect(parsed.input).toEqual(0);
+  });
+
+  test("check optional fields", () => {
+    const data = {
+      time: Date.now(),
+    };
+    const parsed = zStatisticCreate.parse(data);
+    expect(parsed.input).toBeUndefined;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Statistic/Statistic.types.test.ts:46]: `toBeUndefined` not being invoked

File: packages/tcore-sdk/src/Types/Summary/Summary.types.test.ts [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+
+  test("check required fields", () => {
+    const data = {
+      device: 1,
+    };
+    try {
+      zSummary.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.analysis[0]).toBe("Required");
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Summary/Summary.types.test.ts:37]: Check if the other ones are required as well

**Review by matheuslbenachio — CHANGES_REQUESTED** [review]
Fix requested changes so I can merge

File: packages/tcore-sdk/src/Types/Account/Account.types.test.ts [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+  test("error if invalid option", () => {
+    const data = {
+      fields: [" "],
+    };
+    try {
+      zAccountListQuery.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.fields[0]).toBe(
+        "Invalid enum value. Expected 'id' | 'name' | 'username' | 'password' | 'password_hint' | 'created_at'"
```
[the remark this answers is not in the document]
Comment (FabianoEger) [reply packages/tcore-sdk/src/Types/Account/Account.types.test.ts:62]: @cercena use startsWith('Invalid enum value....);

File: packages/tcore-sdk/src/Types/Account/Account.types.test.ts [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+      expect(e.fieldErrors.fields[0].startsWith("Invalid enum value.")).toBeTruthy();
+    }
+  });
+
+  test("assure empty query", () => {
+    const data = {
+      fields: [],
+    };
+    const parsed = zAccountListQuery.parse(data);
+    expect(parsed.fields).toBeNull;
```
Comment (matheuslbenachio) [inline packages/tcore-sdk/src/Types/Account/Account.types.test.ts:70]: This doesn't make sense, this function is not being called. Please review everything like this and then i'll check the tests.

**Review by matheuslbenachio — CHANGES_REQUESTED** [review]

**Pull request merged**